### PR TITLE
Update to Ubuntu Bionic and remove python2

### PR DESCRIPTION
**Pull request by tobias-urdin**
Update to Ubuntu Bionic and remove python2 since we don't use it for testing anymore.

### CODE REVIEW

File: dockerhub/travis-ci-image.dockerfile [hunk 1]
```diff
@@ -49,9 +41,11 @@ ENV LANG en_US.UTF-8
 RUN update-locale
 RUN locale-gen $LANG
 
+RUN curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
```
Comment (jd) [inline dockerhub/travis-ci-image.dockerfile:44]: Hm you should have pip already with Ubuntu, why this?

**Review by jd — CHANGES_REQUESTED** [review]

File: dockerhub/travis-ci-image.dockerfile (hunk 1 above)
Comment (tobias-urdin) [reply dockerhub/travis-ci-image.dockerfile:44]: No, we need to install pip. It's not packaged in either python3-pip or python3.7-pip (third party), those packages does not exist. You don't get it with python3{.7}-dev either. So actually need to install pip for both 3.6 and 3.7 like this.

Comment (tobias-urdin) [reply dockerhub/travis-ci-image.dockerfile:45]: I can show you, let me push a patch.

Comment (tobias-urdin) [reply dockerhub/travis-ci-image.dockerfile:45]: (we need 3.7 pip as well)

Comment (tobias-urdin) [reply dockerhub/travis-ci-image.dockerfile]: We'd get python3.6 pip with python3-pip, but python3.7-pip does not exist in the third party repo. We are testing with 3.7 right now, not sure if we will break 3.7 (no pip?) if we merge this?

**Comment by chungg** [comment]
i don't know if this is 100% right. but i'm ok with this.

File: dockerhub/travis-ci-image.dockerfile (hunk 1 above)
Comment (jd) [reply dockerhub/travis-ci-image.dockerfile]: Let me check, it should be easy to see.

File: dockerhub/travis-ci-image.dockerfile [hunk 2]
```diff
@@ -32,14 +28,12 @@ RUN apt-get update -y && apt-get install -qy \
...
-        python-rados \
+        python3-rados \
         ceph \
 # For prometheus
         libsnappy-dev \
         libprotobuf-dev
 
-# Install Redis from more recent version of Ubuntu to have >= 3.2 and not only 3.0
-RUN echo 'deb http://old-releases.ubuntu.com/ubuntu artful universe' > /etc/apt/sources.list
 RUN apt-get update -y && apt-get install -qy \
```
Comment (jd) [inline dockerhub/travis-ci-image.dockerfile:37]: This should be merged with the `apt-get update/install` above to make the build faster. No reason to have it split anymore.

File: dockerhub/travis-ci-image.dockerfile [hunk 3]
```diff
@@ -50,8 +44,7 @@ RUN update-locale
 RUN locale-gen $LANG
 
 #NOTE(sileht): Upgrade python dev tools
-RUN pip install -U pip tox virtualenv
-RUN pip install zipp configparser
+RUN pip3 install -U pip tox virtualenv
```
Comment (jd) [inline dockerhub/travis-ci-image.dockerfile:47]: If you want to use python3.7 you need to use `python3.7 -m pip`

**Review by jd — CHANGES_REQUESTED** [review]

**Comment by jd** [comment]
I've tested your dockerfile:

```
tester@db53c25c6709:~/src$ python3.7 -m pip

Usage:
  /usr/bin/python3.7 -m pip <command> [options]

[…]
```

👍

**Review by jd — APPROVED** [review]

**Comment by jd** [comment]
That might break the CI so let's be prepared to fix it again. 😬

**Pull request merged**